### DISCUSSION TITLE
Replacements for WP-Plugins

### DISCUSSION THREAD

**Discussion by che-rusker** [GetPublii/Publii#1334] [Ideas]
Howdy,

not for the first time I am thinking about switching my blogs from WP to Publii. 

I am using [footnotes](https://wordpress.org/plugins/footnotes/) a lot, but haven't seen anything for that yet.

Also I'd love to have [image comparison](https://wordpress.org/plugins/image-comparison-block/#developers) in Publii as well. 

I am not a developer, but wouldn't mind editing some files to implement the functionality.
Perhaps [like so](https://www.codehim.com/menu/super-simple-image-comparison-slider-in-html/)?

Would be great if someone with the necessary knowledge could turn this into a plugin.

Cheers,
Michael

**Comment by dziudek** [comment]
Hi,

Both features seems to be quite niche - currently we have no plans for such official plugins. Maybe somebody from the community will built them. Regarding image comparision - you can always use raw HTML mode and implement whatever you need. Also you can consider plugin like: https://marketplace.getpublii.com/plugins/smart-replacer/ to create a custom rule which will get as params 2 URLs to images and then will place them in a proper HTML context with additional HTML syntax and scripts like `[my_image_comparision img1="URL_TO_FIRST_IMAGE" img2="URL_TO_SECOND_IMAGE"]` to avoid writing the code all the time.

**Reply by che-rusker** [reply]
I would not call something that keeps me from switching my 4 blogs (news, food, photography...) with thousands of posts from WP to Publii "niche".

Ah well, at least WP works as desired...

**Comment by candidexmedia** [comment]
Hi Michael,

## Footnotes

> I am using [footnotes](https://wordpress.org/plugins/footnotes/) a lot, but haven't seen anything for that yet.

One of the three Publii editors (the WYSIWYG editor) uses the TinyMCE, and TinyMCE has a [footnotes plugin](https://www.tiny.cloud/docs/tinymce/latest/footnotes/), but it's not free... I'm also not sure which version of TinyMCE Publii uses, and whether that Plugin would be compatible.

In any case, footnotes are possible manually, but it does require a few steps. Here's an example:

![image](https://github.com/GetPublii/Publii/assets/95003834/ea8391c6-3e23-4f9e-8091-9ab8e9977a59)

```html
<p>Heading<sup><a href="#footnotes-2">1</a></sup> is a a heading that makes a lot of sense. That said, only use it once.<sup><a href="#footnotes-3">2</a></sup></p>
<hr>
<section class="footnotes">
	<ol>
		<li id="footnotes-2">This is my footnote</li>
		<li id="footnotes-3">This is another footnote</li>
	</ol>
</section>
```

Result:

![](https://i.imgur.com/Y1GsI9C.gif)


This implementation could also work in theory, but I haven't tested it yet across multiple pages: https://codepen.io/SitePoint/pen/QbMgvY?editors=1100


You may also want to consider a footnote JS library which displays the note in context with the content, such as [Littlefoot](https://littlefoot.js.org/).


## Image Comparison

> Also I'd love to have [image comparison](https://wordpress.org/plugins/image-comparison-block/#developers) in Publii as well.
> 
> I am not a developer, but wouldn't mind editing some files to implement the functionality. Perhaps [like so](https://www.codehim.com/menu/super-simple-image-comparison-slider-in-html/)?

Yup, that's what I'd recommend doing in this case, either with the [code you shared](https://www.codehim.com/menu/super-simple-image-comparison-slider-in-html/) or [another JavaScript library](https://github.com/abelcabezaroman/definitive-image-comparison-slider?tab=readme-ov-file) that can be loaded via hosting the JS files or via CDN.

There are also ways of making the implementation easier and baking it into a custom theme by [extending the WYSIWYG Editor](https://gist.github.com/candidexmedia/367711a80274b4c005fd5983df97d20c).

There isn't official plugins documentation yet, but someone in the former forum started working on some [unofficial plugin docs](https://p4p.pages.dev/tags/plugins/). Maybe those instructions could help you create a plugin, or maybe you could [reach out to that person to collaborate on plugins](https://p4p.pages.dev/contact/).

**Reply by HowToMeetLadies** [reply]
I've prepared something... [Footnotes plugin for Publii](https://p4p.pages.dev/plugin-footnotes)

This plugin will replace `[^n]:` (definition) and `[^n]` (link) with the following markup:

`[^1]:` becomes `<span class="footnote" id="fn-1"><a href="#fnref-1">1</a>.</span>`
and `[^1]` becomes `<sup class="footnote" id="fnref-1"><a href="#fn-1" rel="footnote">1</a></sup>`

You can see an example on the page. 🍻

**Reply by che-rusker** [reply]
Many thanks to anyone coming up with constructive proposals and practical hints. 👍

**Reply by HowToMeetLadies** [reply]
One more way:

If you would like to support Publii, you can also do this with the SmartReplacer plugin.

Then use the following regexes:

Pattern: `/\[\^(\d+)\]:/gi`
Replace with: `<span class="footnote" id="fn-$1"><a href="#fnref-$1">$1</a>.</span>`

Pattern: `/\[\^(\d+)\]/gi`
Replace with: `<sup class="footnote" id="fnref-$1"><a href="#fn-$1" rel="footnote">$1</a></sup>`

**Reply by candidexmedia** [reply]
Chris, you're a gem to the Publii community! This is what open source is all about. 🚀 

I had also tried working on a way to help generate footnotes ([demo](https://editor.p5js.org/candidexmedia/full/8QULxEmZy), [source code](https://editor.p5js.org/candidexmedia/sketches/8QULxEmZy)), but the plugin is a much better idea.

**Reply by Dricc123** [reply]
> Many thanks to anyone coming up with constructive proposals and practical hints. 👍

Has it worked for you? Could you help me with it: https://github.com/GetPublii/Publii/issues/2262

**Comment by HowToMeetLadies** [comment]
🤓 [ImageComparison plugin for Publii](https://p4p.pages.dev/plugin-imagecomparison/)

**Reply by che-rusker** [reply]
Wonderful! Vielen Dank! 🙂 

I'll check you stuff out soon.
